src/ui_parts/viewport.gd

```diff
@@ -57,39 +57,35 @@ func _unhandled_input(event: InputEvent) -> void:
 		zoom_menu.zoom_level *= event.factor
 	
 	if event is InputEventMouseButton and event.is_pressed():
-		var fac = event.factor
-		if fac == roundf(fac): # Detects unsupported device.
-			fac = 1.0
-		if event.ctrl_pressed and event.shift_pressed:
+		var move_vec := Vector2.ZERO
+		var zoom_dir := 0
+		if ( not event.ctrl_pressed and not event.shift_pressed and not GlobalSettings.use_ctrl_for_zoom)\
+		or ( event.ctrl_pressed and GlobalSettings.use_ctrl_for_zoom ):
 			match event.button_index:
-				MOUSE_BUTTON_WHEEL_UP:
-					set_view(view.position - Vector2(fac / zoom * 32, 0.0))
-				MOUSE_BUTTON_WHEEL_DOWN:
-					set_view(view.position + Vector2(fac / zoom * 32, 0.0))
-				MOUSE_BUTTON_WHEEL_LEFT:
-					set_view(view.position - Vector2(0.0, fac / zoom * 32))
-				MOUSE_BUTTON_WHEEL_RIGHT:
-					set_view(view.position + Vector2(0.0, fac / zoom * 32))
-		elif event.ctrl_pressed:
+				MOUSE_BUTTON_WHEEL_UP when GlobalSettings.invert_zoom: zoom_dir -= 1
+				MOUSE_BUTTON_WHEEL_DOWN when GlobalSettings.invert_zoom: zoom_dir += 1
+				MOUSE_BUTTON_WHEEL_UP: zoom_dir += 1
+				MOUSE_BUTTON_WHEEL_DOWN: zoom_dir -= 1
+		elif event.shift_pressed:
 			match event.button_index:
-				MOUSE_BUTTON_WHEEL_UP:
-					set_view(view.position - Vector2(0.0, fac / zoom * 32))
-				MOUSE_BUTTON_WHEEL_DOWN:
-					set_view(view.position + Vector2(0.0, fac / zoom * 32))
-				MOUSE_BUTTON_WHEEL_LEFT:
-					set_view(view.position - Vector2(fac / zoom * 32, 0.0))
-				MOUSE_BUTTON_WHEEL_RIGHT:
-					set_view(view.position + Vector2(fac / zoom * 32, 0.0))
+				MOUSE_BUTTON_WHEEL_UP: move_vec = Vector2.LEFT
+				MOUSE_BUTTON_WHEEL_DOWN: move_vec = Vector2.RIGHT
+				MOUSE_BUTTON_WHEEL_LEFT: move_vec = Vector2.UP
+				MOUSE_BUTTON_WHEEL_RIGHT: move_vec = Vector2.DOWN
 		else:
 			match event.button_index:
-				MOUSE_BUTTON_WHEEL_UP when GlobalSettings.invert_zoom:
-					zoom_menu.zoom_out(fac)
-				MOUSE_BUTTON_WHEEL_UP:
-					zoom_menu.zoom_in(fac)
-				MOUSE_BUTTON_WHEEL_DOWN when GlobalSettings.invert_zoom:
-					zoom_menu.zoom_in(fac)
-				MOUSE_BUTTON_WHEEL_DOWN:
-					zoom_menu.zoom_out(fac)
+				MOUSE_BUTTON_WHEEL_UP: move_vec = Vector2.UP
+				MOUSE_BUTTON_WHEEL_DOWN: move_vec = Vector2.DOWN
+				MOUSE_BUTTON_WHEEL_LEFT: move_vec = Vector2.LEFT
+				MOUSE_BUTTON_WHEEL_RIGHT: move_vec = Vector2.RIGHT
+		var fac = event.factor
```

Comment:
Add static typing here. Also, while at it, let's have 2 spaces between the code and the comment, consistent with the rest of the documentation.

Reply:
Ill fix the spaces before the comment.
Idk how it happened but i think you just reviewed an older version of this PR.
I already added static typing to that.